web/src/pages/ResourcesDashboard.tsx

```diff
@@ -150,9 +157,46 @@ const ResourcesDashboard = () => {
     toast.success(t("message.succeed-copy-resource-link"));
   };
 
+  const handleFetchMoreResourceBtnClick = async () => {
+    try {
+      const fetchedResource = await resourceStore.fetchResourceListWithLimit(DEFAULT_MEMO_LIMIT, resources.length);
+      if (fetchedResource.length < DEFAULT_MEMO_LIMIT) {
+        setIsComplete(true);
+      } else {
+        setIsComplete(false);
+      }
+    } catch (error: any) {
+      console.error(error);
+      toast.error(error.response.data.message);
+    }
+  };
+
+  const loadAllResources = async (resolve: () => void) => {
```

Comment:
Use the callback format because both clear and search need to fetch all resources. Writing the same code twice is inelegant. And in the future, there may be other functions that need to fetch all resources. So I think this abstraction is appropriate.